### PR TITLE
Develop sql

### PR DESCRIPTION
**Pull request by artem-kushal**

### CODE REVIEW

File: data-access/sequelize/repositories/hotel.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        const hotels = await this.model.findAll({
+            where: sequelize.and(...conditions),
+            include: HOTEL_RELATIONSHIP
+        });
+
+        return hotels.map((hotel) => HotelParser.parse(hotel));
+    }
+
+    async create(name, country, city, address, rooms, facilities) {
+        const newHotel = await this.model.create({ name, city, country, address });
```
Comment (artem-kushal) [inline data-access/sequelize/repositories/hotel.js]: add transaction for this actions.

File: data-access/sequelize/repositories/booking.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        });
+
+        return BookingParser.parse(booking);
+    }
+
+    async getAllByUserIdAndFilters(user, isCompleted, city, country, criticalDate) {
+        const today = moment(new Date()).format('YYYY-MM-DD');
+        const requestedDateTimeLimit = moment(new Date(criticalDate)).format('YYYY-MM-DD HH:mm:ss');
+
+        const approvedBookingsQuery = `
```
Comment (artem-kushal) [inline data-access/sequelize/repositories/booking.js]: split to 3 methods

File: data-access/tools/mongoose-model-parsers/hotel.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            dbModel.facilities[0] &&
+            dbModel.facilities[0].baseFacility &&
+            dbModel.facilities[0].baseFacility.name
+                ? dbModel.facilities.map((facility) => ({
+                      price: facility.price,
+                      ...FacilityParser.parse(facility.baseFacility)
+                  }))
+                : [];
+
+        const rooms = dbModel && typeof dbModel.rooms[0] !== 'string' ? dbModel.rooms.map((room) => RoomParser.parse(room)) : [];
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply data-access/tools/mongoose-model-parsers/hotel.js]: +

File: data-access/sequelize/repositories/hotel.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        });
+
+        return HotelParser.parse(hotel);
+    }
+
+    async getByFilters(city, country, capacity, dateFrom, dateTo, criticalDate) {
+        const conditions = [];
+
+        if (dateTo && dateFrom) {
+            const requestedDateTimeLimit = moment(new Date(criticalDate)).format('YYYY-MM-DD HH:mm:ss');
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply data-access/sequelize/repositories/hotel.js]: +

File: data-access/sequelize/repositories/booking.js (hunk 2 above)
Comment (formalinfigs) [reply data-access/sequelize/repositories/booking.js]: +

File: business-logic/services/hotel-service.js [hunk 5]
```diff
@@ -12,14 +13,18 @@ class HotelService extends BaseService {
...
     async get(id) {
-        const hotel = await this._repository.get(id);
+        try {
+            const hotel = await this._repository.get(id);
 
-        const { photos, ...parsedHotel } = this._model.parse(hotel);
-        return parsedHotel;
+            return this._model.parse(hotel);
+        } catch (err) {
+            throw { message: 'Hotel not found, invalid id', statusCode: 404 };
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/services/hotel-service.js]: +

**Pull request merged**